test/test_schemas.py

```diff
@@ -74,6 +74,8 @@ def test_example_avro(self):
                             schema.retrieve_alerts(f)
                     else:
                         retrieved_schema, alerts = schema.retrieve_alerts(f)
+                        schema.definition.pop('__named_schemas')
```

Comment:
I worry a little bit about if this is too fragile to future fastavro changes--it's a private data structure, presumably.

What if we check for it something like this:

```
private_fastavro_keys = ['__named_schemas', '__fastavro_parsed']
for key in private_fastavro_keys:
    if key in schema.definition:
        schema.definition.pop(key)
```

Reply:
Alright, I'll add that in. I was worried about that, but I like your solution.

Reply:
Maybe drop any key that starts with double underscore? I have no idea if you expect valid schema content like that though.

Reply:
We'd have to make it more robust then that as it would drop __len__ as well. I also am not sure if any other keys would be added, since it looks like the two added keys are fastavro defaults.

Reply:
Tim's suggestion would also work--I wouldn't expect us to provide any native schema content starting that way.

Reply:
Yup, needed to exclude `__len__` otherwise it ended up in a destructive recursive loop.

Reply:
New code is the following. Have to seperate out the keys into a list otherwise it crashes due to dictionary size changing on the fly.

``` 
fastavro_keys = list(fully_resolved_schema.keys())
for key in fastavro_keys:
    if '__' in key and '__len__' not in key:
    fully_resolved_schema.pop(key)
```